### PR TITLE
grouped indexing

### PR DESCRIPTION
**Pull request by jeroiraz**
Signed-off-by: Jeronimo Irazabal <jeronimo.irazabal@gmail.com>

### CODE REVIEW

File: embedded/store/indexer.go [hunk 1]
```diff
@@ -77,30 +85,71 @@ var (
...
+		WithMaxNodeSize(opts.IndexOpts.MaxNodeSize).
+		WithMaxKeySize(opts.MaxKeyLen).
+		WithMaxValueSize(lszSize + offsetSize + sha256.Size + sszSize + maxTxMetadataLen + sszSize + maxKVMetadataLen). // indexed values
+		WithNodesLogMaxOpenedFiles(opts.IndexOpts.NodesLogMaxOpenedFiles).
+		WithHistoryLogMaxOpenedFiles(opts.IndexOpts.HistoryLogMaxOpenedFiles).
+		WithCommitLogMaxOpenedFiles(opts.IndexOpts.CommitLogMaxOpenedFiles).
+		WithRenewSnapRootAfter(opts.IndexOpts.RenewSnapRootAfter).
+		WithCompactionThld(opts.IndexOpts.CompactionThld).
+		WithDelayDuringCompaction(opts.IndexOpts.DelayDuringCompaction)
+
```
Comment (arriqaaq) [inline embedded/store/indexer.go:113]: I think the part to validate indexopts is missing? 
This line I mean err = indexOpts.Validate()

Comment (jeroiraz) [reply embedded/store/indexer.go:113]: actually, the opts are validated by the btree itself

File: embedded/store/indexer.go [hunk 2]
```diff
@@ -371,78 +420,100 @@ func (idx *indexer) doIndexing() {
...
+			idx._kvs[indexableEntries].V = b[:o]
+			idx._kvs[indexableEntries].T = txID + uint64(i)
 
-		indexableEntries++
+			indexableEntries++
+		}
+
+		bulkSize++
+
+		if bulkSize < idx.maxBulkSize {
```
Comment (arriqaaq) [inline embedded/store/indexer.go:493]: Wouldn't this condition always be true?

File: embedded/store/indexer.go [hunk 3]
```diff
@@ -371,78 +420,100 @@ func (idx *indexer) doIndexing() {
...
+			break
+		}
+		if err != nil {
+			return err
+		}
 	}
 
+	var err error
+
 	if indexableEntries == 0 {
```
Comment (arriqaaq) [inline embedded/store/indexer.go:507]: Can you just add a comment on why ts is being increased when indexableEntries ==0? :)

**Review by arriqaaq — COMMENTED** [review]
minor comments, lgtm

File: embedded/store/indexer.go (hunk 2 above)
Comment (jeroiraz) [reply embedded/store/indexer.go:493]: it shouldn't hold in the last iteration

File: embedded/store/indexer.go (hunk 3 above)
Comment (jeroiraz) [reply embedded/store/indexer.go:507]: when a transaction didn't have any indexable entry, the logical time in the tree was increased. But that is actually not required anymore, given the transaction id is specified per entry now

**Review by arriqaaq — APPROVED** [review]

**Pull request merged**